### DISCUSSION TITLE
Traffic not encrypted when ipsec transport mode configed on the Router

### DISCUSSION THREAD

**Discussion by qianguozheng** [strongswan/strongswan#1671] [Q&A] [answered]
Hi, 
I have a scenario, PC->Router(A) <->Router(B).
Router A have successfully setup the ipsec transport connection to Router B, and ping from Router A to Router B is ok, capture packet between them , found they communicate with esp packet, it's ok.
However, When I ping Router B from PC, it goto Router without esp encryption, I have already set NAT on Router A, the source ip from Router A is itself. That's not right.
I have done some research to find that the xfrm policy setup maybe not right, can anyone help introduce below message meaning ? Thanks

**PC ip 192.168.1.2
Router(A) ip 100.100.100.2
Router(B) ip 100.100.100.1**

/ # ip xfrm policy
src 100.100.100.2/32 dst 100.100.100.1/32 
        dir out priority 367231 ptype main 
        tmpl src 0.0.0.0 dst 0.0.0.0
                proto esp spi 0xcd004040 reqid 1 mode transport
src 100.100.100.1/32 dst 100.100.100.2/32 
        dir in priority 367231 ptype main 
        tmpl src 0.0.0.0 dst 0.0.0.0
                proto esp reqid 1 mode transport
src 0.0.0.0/0 dst 0.0.0.0/0 
        socket in priority 0 ptype main 
src 0.0.0.0/0 dst 0.0.0.0/0 
        socket out priority 0 ptype main 
src 0.0.0.0/0 dst 0.0.0.0/0 
        socket in priority 0 ptype main 
src 0.0.0.0/0 dst 0.0.0.0/0 
        socket out priority 0 ptype main 
src ::/0 dst ::/0 
        socket in priority 0 ptype main 
src ::/0 dst ::/0 
        socket out priority 0 ptype main 
src ::/0 dst ::/0 
        socket in priority 0 ptype main 
src ::/0 dst ::/0 
        socket out priority 0 ptype main 

/ # ip xfrm state
src 100.100.100.2 dst 100.100.100.1
        proto esp spi 0xcd004040 reqid 1 mode transport
        replay-window 0 
        auth-trunc hmac(sha1) 0x60423a57b4f659408a0c61f8d4295f219726b70b 96
        enc cbc(aes) 0xfd227c30807f6f8ce661dcc4ac18610700a61a019ebe0502f05b0748f9e73a14
        sel src 100.100.100.2/32 dst 100.100.100.1/32 
src 100.100.100.1 dst 100.100.100.2
        proto esp spi 0xc35914c9 reqid 1 mode transport
        replay-window 32 
        auth-trunc hmac(sha1) 0x94a1ff2e0e839f24cee3e18c8d40bbb06692aa0c 96
        enc cbc(aes) 0xa7ba95edd667452946624b7414f1ae43291d1ed76a858e2634b9e672661dd886
        sel src 100.100.100.1/32 dst 100.100.100.2/32

**Comment by tobiasbrunner** [comment, marked as answer]
> I have already set NAT on Router A, the source ip from Router A is itself. That's not right.

What do you mean? The IPsec policy only protects traffic between `100.100.100.2` and `100.100.100.1`. Unless you use tunnel mode (or some sort of overlay tunneling protocol), you definitely have to NAT the traffic from `192.168.1.2` to `100.100.100.2` in order to match the policy and protect it. Check your firewall rules and maybe trace the traffic.

**Reply by qianguozheng** [reply]
All as you said, I have NAT the traffic from 192.168.1.2 to 100.100.100.2, and capture the packet on wan interface, the ip is indeed changed to 100.100.100.2. 
But I am not sure the policy is right or not, can you help take a look on that, meanwhile, I will trace the traffic to see where it is going.
Thanks

/ # ip xfrm policy
**src 100.100.100.2/32 dst 100.100.100.1/32
dir out priority 367231 ptype main
tmpl src 0.0.0.0 dst 0.0.0.0
proto esp spi 0xcd004040 reqid 1 mode transport
src 100.100.100.1/32 dst 100.100.100.2/32
dir in priority 367231 ptype main
tmpl src 0.0.0.0 dst 0.0.0.0
proto esp reqid 1 mode transport
src 0.0.0.0/0 dst 0.0.0.0/0**
socket in priority 0 ptype main
src 0.0.0.0/0 dst 0.0.0.0/0
socket out priority 0 ptype main
src 0.0.0.0/0 dst 0.0.0.0/0
socket in priority 0 ptype main
src 0.0.0.0/0 dst 0.0.0.0/0
socket out priority 0 ptype main
src ::/0 dst ::/0
socket in priority 0 ptype main
src ::/0 dst ::/0
socket out priority 0 ptype main
src ::/0 dst ::/0
socket in priority 0 ptype main
src ::/0 dst ::/0
socket out priority 0 ptype main

**Reply by tobiasbrunner** [reply]
> But I am not sure the policy is right or not, can you help take a look on that

What are you not sure about?

**Reply by qianguozheng** [reply]
Thanks, I have traced the packet flow, and found that packet using traffic selector earlier than do the nat related things
